zerver/lib/send_email.py

```diff
@@ -214,7 +215,12 @@ def send_email(
     template = template_prefix.split("/")[-1]
     logger.info("Sending %s email to %s", template, mail.to)
 
-    if mail.send() == 0:
+    try:
+        sent = mail.send()
+        if sent == 0:
+            logger.error("Error sending %s email to %s", template, mail.to)
+            raise EmailNotDeliveredException
+    except smtplib.SMTPException:
         logger.error("Error sending %s email to %s", template, mail.to)
         raise EmailNotDeliveredException
```

Comment:
This would also be good to add a test for -- it should be pretty doable to just mock `mail.send` with a `side_effect` of the appropriate exception class.